backend/src/main/java/com/gleam/backend/api/ui/ApiController.java

```diff
@@ -0,0 +1,41 @@
+package com.gleam.backend.api.ui;
+
+import com.gleam.backend.common.dto.RegisterUserRequest;
+import com.gleam.backend.common.dto.UserDto;
+import com.gleam.backend.common.event.GetOrCreateUserEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+@RestController
+@RequestMapping("/api")
+public class ApiController {
+    private final ApplicationEventPublisher eventPublisher;
+
+    public ApiController(ApplicationEventPublisher eventPublisher) {
+        this.eventPublisher = eventPublisher;
+    }
+
+    @PostMapping("/getOrCreateUser")
+    public ResponseEntity<UserDto> getOrCreateUser(@RequestBody RegisterUserRequest requestDto) {
+        var userFuture = new CompletableFuture<UserDto>();
+        var event = new GetOrCreateUserEvent(requestDto, userFuture);
+        eventPublisher.publishEvent(event);
+
+        try {
+            var userDto = userFuture.get(5, TimeUnit.SECONDS);
```

Comment:
Mit Futures habe ich noch nicht gearbeitet... Ich vermute du bekommst hier das Objekt zurück mit einem Timeout von 5 Sekunden?

Reply:
Korrekt, das Objekt kommt dabei möglichst instant zurück - wenn man .get verwendet aber nicht asynchron, d.h. der main thread wird blockiert und spätestens nach 5 Sekunden released. 
Das ist nicht was ich möchte, hatte zugegebenermaßen angenommen, dass .get schon async läuft, mich aber auch eben selbst gefragt wie das ohne async annotations funktionieren sollte.

Die Futures sind allgemein aber auf concurrency ausgelegt, das heißt ein Umbau wäre einfach, würde diesen PR aber sprengen. Gehe ich in einem separaten PR an.